src/test/java/org/broadinstitute/hellbender/tools/exome/segmentation/AlleleFractionSegmenterUnitTest.java

```diff
@@ -91,36 +79,59 @@ public void testChromosomesOnDifferentSegments() {
         positions.addAll(CopyRatioSegmenterUnitTest.randomPositions("chr3", chainLength, random, trueMemoryLength/4));
 
         final int trueState = 2;    //fix everything to the same state 2
+        final List<Double> minorAlleleFractionSequence = Collections.nCopies(positions.size(), trueMinorAlleleFractions[trueState]);
+
+        final AllelicCountCollection counts = generateCounts(minorAlleleFractionSequence, positions, random, trueParams);
+
+        final AlleleFractionSegmenter segmenter = new AlleleFractionSegmenter(10, counts, AllelicPanelOfNormals.EMPTY_PON);
+        final TargetCollection<AllelicCount> tc = new HashedListTargetCollection<>(counts.getCounts());
+        final List<Pair<SimpleInterval, Double>> segmentation = segmenter.findSegments();
+        final List<ModeledSegment> segments = segmentation.stream()
+                .map(pair -> new ModeledSegment(pair.getLeft(), tc.targetCount(pair.getLeft()), pair.getRight()))
+                .collect(Collectors.toList());
+
+        //check that each chromosome has at least one segment
+        final int numDifferentContigsInSegments = (int) segments.stream().map(ModeledSegment::getContig).distinct().count();
+        Assert.assertEquals(numDifferentContigsInSegments, 3);
+    }
 
+    //visible for testing joint segmentation
+    protected static AllelicCountCollection generateCounts(final List<Double> minorAlleleFractionSequence,
+                                                  final List<SimpleInterval> positions,
+                                                  final Random random,
+                                                  final AlleleFractionGlobalParameters trueParams) {
         //translate to ApacheCommons' parametrization of the gamma distribution
-        final double gammaShape = trueParams.getMeanBias() * trueParams.getMeanBias() / trueParams.getBiasVariance();
-        final double gammaScale = trueParams.getBiasVariance() / trueParams.getMeanBias();
-        final RandomGenerator rng = RandomGeneratorFactory.createRandomGenerator(new Random(563));
-        final GammaDistribution biasGenerator = new GammaDistribution(rng, gammaShape, gammaScale);
+        final GammaDistribution biasGenerator = getGammaDistribution(trueParams);
+        final double outlierProbability = trueParams.getOutlierProbability();
 
         final AllelicCountCollection counts = new AllelicCountCollection();
-        for (int n = 0; n < positions.size(); n++) {
-            final int numReads = 100;
-            final double minorFraction = trueMinorAlleleFractions[trueState];
-            final double bias = biasGenerator.sample();
 
-            //flip a coin to decide alt minor (alt fraction = minor fraction) or ref minor (alt fraction = 1 - minor fraction)
-            final double altFraction =  random.nextDouble() < 0.5 ? minorFraction : 1 - minorFraction;
+        for (int n = 0; n < minorAlleleFractionSequence.size(); n++) {
```

Comment:
It's not worth it to me here because you still have to construct an empty `AllelicCountCollection`.